packages/mobile/src/store/offline-downloads/slice.ts

```diff
@@ -92,24 +92,45 @@ const slice = createSlice({
       { payload: trackIds }: PayloadAction<string[]>
     ) => {
       trackIds.forEach((trackId) => {
-        state.downloadStatus[trackId] = OfflineDownloadStatus.INIT
+        if (
+          !state.downloadStatus[trackId] ||
+          (state.downloadStatus[trackId] !== OfflineDownloadStatus.LOADING &&
+            state.downloadStatus[trackId] !== OfflineDownloadStatus.SUCCESS)
+        ) {
+          state.downloadStatus[trackId] = OfflineDownloadStatus.INIT
```

Comment:
We could also fix this issue by adding conditions to the sagas/functions that call these actions. But this was the least complicated path.